examples/domainmodel/test/cross-refs.test.ts

```diff
@@ -0,0 +1,58 @@
+/******************************************************************************
+ * Copyright 2021 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import { AstNode, LangiumDocument, ReferenceDescription } from "langium";
+import { DefaultIndexManager } from "langium/lib/index/index-manager";
+import { parseDocument } from "langium/lib/test";
+import { createDomainModelServices } from "../src/language-server/domain-model-module";
+import { Domainmodel } from "../src/language-server/generated/ast";
+
+const services = createDomainModelServices()
+
+const datatypeFile = `
+datatype String
+datatype Int
+datatype Decimal
+
+package big {
+    datatype Int
+    datatype Decimal
+}
+`;
+
+const referencingFile = `
+ entity Blog {
+    title: String
+    description: String
+    time: big.Int
+}
+`;
+describe('Cross references', () => {
+    const datatypeDoc: LangiumDocument = parseDocument(services, datatypeFile);
```

Comment:
Shall we add a type parameter `<T = AstNode>` to LangiumDocument and also to `parseDocument` so we can avoid typecasting further below? This line could then be written as
```typescript
const datatypeDoc = parseDocument<DomainModel>(services, datatypeFile);
```
and `datatypeDoc` would have the inferred type `LangiumDocument<DomainModel>`.

Reply:
Would be cool!